agents/gotToGoodBraitenberg.py

```diff
@@ -0,0 +1,71 @@
+from animalai.envs.actions import AAIActions, AAIAction
+from animalai.envs.raycastparser import RayCastParser
+from animalai.envs.raycastparser import RayCastObjects
+
+class Braitenberg():
+    """Implements a simple Braitenberg vehicle agent that heads towards food
+    Can change the number of rays but only responds to GOODGOALs, GOODGOALMULTI and BADGOAL"""
+    def __init__(self, no_rays):
+        self.no_rays = no_rays
+        assert(self.no_rays % 2 == 1), "Only supports odd number of rays (but environment should only allow odd number"
+        self.listOfObjects = [RayCastObjects.GOODGOAL, RayCastObjects.GOODGOALMULTI, RayCastObjects.BADGOAL, RayCastObjects.IMMOVABLE, RayCastObjects.MOVABLE]
+        self.raycast_parser = RayCastParser(self.listOfObjects, self.no_rays)
+        self.actions = AAIActions()
+        self.prev_action = self.actions.NOOP
+
+    def prettyPrint(self, obs) -> str:
+        """Prints the parsed observation"""
+        return self.raycast_parser.prettyPrint(obs)
+    
+    def get_action(self, obs) -> AAIAction:
+        """Returns the action to take given the current parsed raycast observation"""
+        obs = self.raycast_parser.parse(obs)
+        newAction = self.actions.NOOP
+        if self.ahead(obs, RayCastObjects.GOODGOALMULTI):
+            newAction = self.actions.FORWARDS
+        elif self.left(obs, RayCastObjects.GOODGOALMULTI):
+            newAction = self.actions.FORWARDSLEFT
+        elif self.right(obs, RayCastObjects.GOODGOALMULTI):
+            newAction = self.actions.FORWARDSRIGHT
+        elif self.ahead(obs, RayCastObjects.GOODGOAL):
+            newAction = self.actions.FORWARDS
+        elif self.left(obs, RayCastObjects.GOODGOAL):
+            newAction = self.actions.FORWARDSLEFT
+        elif self.right(obs, RayCastObjects.GOODGOAL):
+            newAction = self.actions.FORWARDSRIGHT
+        elif self.ahead(obs, RayCastObjects.BADGOAL):
+            newAction = self.actions.BACKWARDS
+        elif self.left(obs, RayCastObjects.BADGOAL):
+            newAction = self.actions.BACKWARDSLEFT
+        elif self.right(obs, RayCastObjects.BADGOAL):
+            newAction = self.actions.BACKWARDSRIGHT
+        else:
+            if self.prev_action == self.actions.NOOP or self.prev_action == self.actions.BACKWARDS:
+                newAction = self.actions.LEFT
+            else:
+                newAction = self.prev_action    
```

Comment:
Just to double check again: It is intentional to keep using the previous action when e.g. the goal goes out of sight? E.g. you want to keep doing forwardsleft, instead of doign forwardsleft once and then forwards?

Reply:
Yeah, for now. It's kind of arbitrary here, as it is just an example for more sophisticated/well-thought-about agents later. This agent is copied from the examples folder (I didn't write it).